### PR TITLE
Several improvements / fixes for functions

### PR DESCRIPTION
**Pull request by rzubek**
New: function arguments can now be typed arrays, in addition to string, number, etc. Addresses issue #304 and part of issue #317. See AddArrayParameters().

Fix: error messages coming from the server caused exceptions to be thrown during deserialization; this is because the message field was coming back as an array but the deserializer expected a string.

Merged: PR #312. This code builds on PR 312 so it was incorporated.

### CODE REVIEW

**Comment by retendo** [comment]
Would be great to have something like this merged 👍

Question:
Wouldn't it be more helpful if `items` is just this:
`public IList<FunctionParameters>? Items { get; set; }`

**Comment by remixtedi** [comment]
When are you planning to merge this?

**Comment by rzubek** [comment]
> When are you planning to merge this?

If you're asking me, it's not my repo, I don't have any permissions here. This is waiting for @kayhantolga to review and merge.

**Comment by kayhantolga** [comment]
Hi 
this is a great improvement, Thanks! Could you please update the code to target https://github.com/betalgo/openai/tree/feature/functionCall branch?
I moved some of the files and made some changes.

File: OpenAI.SDK/ObjectModels/ResponseModels/BaseResponse.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ public record Error
 {
     [JsonPropertyName("code")] public string? Code { get; set; }
 
-    [JsonPropertyName("message")] public string? Message { get; set; }
```
Comment (kayhantolga) [inline OpenAI.SDK/ObjectModels/ResponseModels/BaseResponse.cs:16]: Why do we need this? Do you have any sample responses that show why the change is necessary?

Comment (rzubek) [reply OpenAI.SDK/ObjectModels/ResponseModels/BaseResponse.cs:16]: Yes, please see the sample response in the description of commit https://github.com/betalgo/openai/pull/320/commits/80bde477c1b1c23077d5e9f20461b1ebd724f736

It came back as an array of strings (rather than a string) which caused the JSON serializer to blow up

**Comment by kayhantolga** [comment]
@rzubek do you think you can do the changes anytime soon? I want to include your changes into 7.2.0
